onnx/common/tensor.h

```diff
@@ -222,49 +217,25 @@ struct Tensor final {
   bool is_raw_data() const {
     return is_raw_data_;
   }
-
-  // this += a
-  // Supported for
-  // FLOAT, BOOL, INT8, INT16, INT32, UINT8, UINT16, INT64,
-  // UINT32, UINT64, DOUBLE,
-  // TODO: Support for FLOAT16, COMPLEX64, COMPLEX128
-  void add(const Tensor& a);
-
-  // this -= a
-  // Supported for
-  // FLOAT, BOOL, INT8, INT16, INT32, UINT8, UINT16, INT64,
-  // UINT32, UINT64, DOUBLE
-  // TODO: Support for FLOAT16, COMPLEX64, COMPLEX128
-  void subtract(const Tensor& a);
-
-  // this *= a
-  // Supported for
-  // FLOAT, BOOL, INT8, INT16, INT32, UINT8, UINT16, INT64,
-  // UINT32, UINT64, DOUBLE
-  // TODO: Support for FLOAT16, COMPLEX64, COMPLEX128
-  void multiply(const Tensor& a);
-
-  // this /= a
-  // Supported for
-  // FLOAT, INT8, INT16, INT32, UINT8, UINT16, INT64,
-  // UINT32, UINT64, DOUBLE
-  // TODO: Support for FLOAT16, COMPLEX64, COMPLEX128
-  void divide(const Tensor& a);
-
-  // Element-wise square root of This
-  // Supported for
-  // FLOAT, DOUBLE,
-  // TODO: Support for FLOAT16
-  void sqrt();
-
-  // Element wise scaling of tensor s
-  // s is one dimensional, has size M, where M is size of first dimension of tensor
-  // s must have has data type corresponding to this
-  // Supported for
-  // FLOAT16, FLOAT, DOUBLE
-  void scale_by_first_dim(const Tensor& s);
```

Comment:
I personally highly agree to remove these functions. They were written because of some specific passes in onnx optimizer, so only a confusing subset of operations (like `scale_by_first_dim`) was implemented. Now onnx optimizer decides to not use these operation implementations any more (for more context please see https://github.com/onnx/optimizer/issues/9). However, I don't know whether there are libraries other than onnx-optimizer using these functions.

Could you please share your opinions? @askhade @gramalingam @jcwchen

Reply:
Thanks for bringing this up. I am fine with removing them and I am not aware any other libraries using these functions.

Reply:
Sounds fine to me too, thanks!